src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -55,6 +55,30 @@ public static void setUpClass() throws IOException {
         testData = new TestUtils.TestData(testIndexVectors.getPath(), testQueries.getPath());
     }
 
+    /**
+     * This test performs basic validation steps such as index, search in order to verify that kNN works with Segment
+     * replication.
+     * @throws Exception
+     */
+    public void testSegmentReplication() throws Exception {
+        String field1 = "field1";
+        String field2 = "field2";
+        createKnnIndex(INDEX_NAME, getKNNSegrepIndexSettings(), createKnnIndexMapping(Arrays.asList(field1, field2), Arrays.asList(2, 2)));
+
+        Float[] vector = { 6.0f, 7.0f };
```

Comment:
better use different vectors for each doc, it should be fine as you're not checking scores, just overall count